backend/postgres/postgres.go

```diff
@@ -0,0 +1,955 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	_ "embed"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/microsoft/durabletask-go/api"
+	"github.com/microsoft/durabletask-go/backend"
+	"github.com/microsoft/durabletask-go/internal/helpers"
+	"github.com/microsoft/durabletask-go/internal/protos"
+
+	"google.golang.org/protobuf/encoding/protojson"
+
+	_ "github.com/jackc/pgx/v5/stdlib"
```

Comment:
Although it's a bit more of a change, I recommend using pgx's native APIs and not the stdlib wrapper.

1. The authors of pgx recommend using the stdlib wrapper only when you're trying to use the same code for multiple databases.
2. The native driver has much better performance.
3. The native driver offers some additional things (see my other comments below)